internal/controller/k8s_resources_test.go

```diff
@@ -616,3 +635,249 @@ func (suite *ControllerTestSuite) TestExtractObjectMetas() {
 		})
 	}
 }
+
+func (suite *ControllerTestSuite) TestConvertIngressToStringList() {
+	tests := map[string]struct {
+		ingress        []v1.LoadBalancerIngress
+		expectedResult []string
+	}{
+		"both IP and Hostname": {
+			ingress: []v1.LoadBalancerIngress{
+				{IP: "192.168.1.1", Hostname: "example.com"},
+			},
+			expectedResult: []string{"192.168.1.1", "example.com"},
+		},
+		"only IP": {
+			ingress: []v1.LoadBalancerIngress{
+				{IP: "192.168.1.2"},
+			},
+			expectedResult: []string{"192.168.1.2"},
+		},
+		"only Hostname": {
+			ingress: []v1.LoadBalancerIngress{
+				{Hostname: "another-example.com"},
+			},
+			expectedResult: []string{"another-example.com"},
+		},
+		"empty ingress": {
+			ingress:        []v1.LoadBalancerIngress{},
+			expectedResult: []string{},
+		},
+		"nil IP and Hostname": {
+			ingress: []v1.LoadBalancerIngress{
+				{IP: "", Hostname: ""},
+			},
+			expectedResult: []string{},
+		},
+	}
+
+	for name, tt := range tests {
+		suite.Run(name, func() {
+			result := convertIngressToStringList(tt.ingress)
+			assert.Equal(suite.T(), tt.expectedResult, result)
+		})
+	}
+}
+
+func (suite *ControllerTestSuite) TestConvertServicePortsToPorts() {
+
+	tests := map[string]struct {
```

Comment:
I'd like to see another test case for the case where `Protocol` is `""`, which should default to `"TCP"`.